### PR TITLE
Fixing "Redundant specification of type arguments errors"

### PR DESCRIPTION
**Pull request by iloveeclipse**
See https://github.com/eclipse-platform/eclipse.platform.ui/issues/1267

### CODE REVIEW

**Comment by iloveeclipse** [comment]
Converted to a draft since we have to understand if removing type arguments would be safe in the light of the possible JDT regression / fix for the remaining issue.

File: tools/bundles/org.eclipse.e4.tools.emf.ui/src/org/eclipse/e4/tools/emf/ui/internal/common/component/MenuItemEditor.java [hunk 1]
```diff
@@ -265,15 +265,15 @@ public void widgetSelected(SelectionEvent e) {
...
 				public ItemType convert(Boolean fromObject) {
 					return null;
 				}
 			});
 			final UpdateValueStrategy<ItemType, Boolean> m2t = new UpdateValueStrategy<>();
-			m2t.setConverter(new Converter<ItemType, Boolean>(ItemType.class, boolean.class) {
+			m2t.setConverter(new Converter<>(ItemType.class, boolean.class) {
 
 				@Override
 				public Boolean convert(ItemType fromObject) {
```
Comment (srikanth-sankaran) [inline tools/bundles/org.eclipse.e4.tools.emf.ui/src/org/eclipse/e4/tools/emf/ui/internal/common/component/MenuItemEditor.java:279]: I have verified manually that eliding the type arguments in these cases and opting for <> is correct.

File: tools/bundles/org.eclipse.e4.tools.emf.ui/src/org/eclipse/e4/tools/emf/ui/internal/common/component/ToolItemEditor.java [hunk 2]
```diff
@@ -287,14 +287,14 @@ public void widgetSelected(SelectionEvent e) {
...
 				public ItemType convert(Boolean fromObject) {
 					return null;
 				}
 			});
 			final UpdateValueStrategy<ItemType, Boolean> m2t = new UpdateValueStrategy<>();
-			m2t.setConverter(new Converter<ItemType, Boolean>(ItemType.class, boolean.class) {
+			m2t.setConverter(new Converter<>(ItemType.class, boolean.class) {
 
 				@Override
 				public Boolean convert(ItemType fromObject) {
```
Comment (srikanth-sankaran) [inline tools/bundles/org.eclipse.e4.tools.emf.ui/src/org/eclipse/e4/tools/emf/ui/internal/common/component/ToolItemEditor.java:300]: I have verified that these type arguments when elided and <> operator is opted for, produce identical types

File: tools/bundles/org.eclipse.e4.tools.emf.ui/src/org/eclipse/e4/tools/emf/ui/internal/common/component/PlaceholderEditor.java [hunk 3]
```diff
@@ -198,14 +198,14 @@ private Composite createForm(Composite parent, final EMFDataBindingContext conte
...
 				public MUIElement convert(String fromObject) {
 					return null;
 				}
 			});
 			final UpdateValueStrategy<MUIElement, String> m2t = new UpdateValueStrategy<>();
-			m2t.setConverter(new Converter<MUIElement, String>(MUIElement.class, String.class) {
+			m2t.setConverter(new Converter<>(MUIElement.class, String.class) {
 
 				@Override
 				public String convert(MUIElement fromObject) {
```
Comment (srikanth-sankaran) [inline tools/bundles/org.eclipse.e4.tools.emf.ui/src/org/eclipse/e4/tools/emf/ui/internal/common/component/PlaceholderEditor.java:211]: These type arguments when elided get inferred to be the same types

**Review by srikanth-sankaran — APPROVED** [review]

**Comment by iloveeclipse** [comment]
OK, thanks @srikanth-sankaran ! We can merge, right?

**Comment by srikanth-sankaran** [comment]
> OK, thanks @srikanth-sankaran ! We can merge, right?

Yes, I believe so.

**Pull request merged**